run

```diff
@@ -658,14 +658,17 @@ build() {
     build_path="build/$1"
   fi
 
+  # Create data path outside build directory
+  data_path="${build_path}/../data"
```

Comment:
As currently written, it looks like this would change HoloHub behavior so that `data` is a subdirectory of `build` rather than at the top level. I.e., `data_path=/workspace/holohub/build/myapp/../data`. I'm not necessarily opposed to that change in behavior, just want to make sure it's known.

I do suggest we run `mkdir` to error out if the relative path does not exist or encounters permissions issues:
```suggestion
  data_path="${build_path}/../data"
  mkdir -p data_path
```

Reply:
Ahh, that is not necessarily what I wanted to do, I wanted to try to have this change be a no-op if a build_path isn't passed. So `data_path="${build_path}/../../data"` is actually the behavior I wanted. Do you see any issues with this?

Reply:
One thing I'm not clear on is whether the data path is required to be outside of the build directory or not. Maybe @jjomier or @wendell-hom could weigh in on what they had in mind with "--buildpath" and build structure conventions?

I'm wondering if it would be less complicated to add a "--datapath" override specifier than to set our own convention for how it relates to a custom "--buildpath"?

Reply:
The data path can be anywhere but we do set it to a default when using the run script.
@NigelNelson what is the use case you are trying to solve here?

Reply:
Following up here to move forward, to solve for the general case while avoiding too many implicit conventions, I suggest that we continue to use the default data path and add a `--datapath` CLI override argument to allow the user to specify a different argument for their specific needs.